noir-projects/noir-contracts/contracts/test_log_contract/src/main.nr

```diff
@@ -107,20 +60,24 @@ contract TestLog {
         let msg_sender_ivpk_m = header.get_ivpk_m(&mut context, context.msg_sender());
         let msg_sender_ovpk_m = header.get_ovpk_m(&mut context, context.msg_sender());
 
+        let event0 = ExampleEvent0 { value0: preimages[0], value1: preimages[1] };
+
         context.encrypt_and_emit_event(
             randomness[0],
-            ExampleEvent0::selector().to_field(),
+            ExampleEvent0::get_event_type_id().to_field(),
```

Comment:
With the constrained encryption we should be able to ged rid of this 👀 should be nice